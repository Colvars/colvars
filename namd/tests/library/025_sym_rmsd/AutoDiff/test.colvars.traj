# step         rmsd                   rmsd_sym               rmsd_min              
           0    3.03766674899195e+00   2.90366645938124e+00   2.90366645938124e+00 
           1    2.65583548116796e+00   2.49357076354985e+00   2.49357076354985e+00 
           2    1.74533483223524e+00   1.47077556782424e+00   1.47077556782424e+00 
           3    1.40555434624453e+00   1.03079831602729e+00   1.03079831602729e+00 
           4    2.06648653553550e+00   1.82627061136782e+00   1.82627061136782e+00 
           5    2.54017219029394e+00   2.33735720111240e+00   2.33735720111240e+00 
           6    2.44511763951316e+00   2.20810814129124e+00   2.20810814129124e+00 
           7    1.90917369343023e+00   1.54370117101215e+00   1.54370117101215e+00 
           8    1.46846059624425e+00   8.57092838197605e-01   8.57092838197605e-01 
           9    1.73306101411381e+00   1.20805207461003e+00   1.20805207461003e+00 
          10    2.23122614737732e+00   1.83168666851452e+00   1.83168666851452e+00 
          11    2.40879160755837e+00   2.03136190092487e+00   2.03136190092487e+00 
          12    2.16660397964880e+00   1.72654978561155e+00   1.72654978561155e+00 
          13    1.72124086387256e+00   1.10943093382589e+00   1.10943093382589e+00 
          14    1.54507113594670e+00   8.13315046570531e-01   8.13315046570531e-01 
          15    1.84201976466439e+00   1.30057662388302e+00   1.30057662388302e+00 
          16    2.18866124657311e+00   1.76429607011380e+00   1.76429607011380e+00 
          17    2.26240711868798e+00   1.85743190829637e+00   1.85743190829637e+00 
          18    1.99090416866217e+00   1.51280845285182e+00   1.51280845285182e+00 
          19    1.59604958772556e+00   9.31109046188012e-01   9.31109046188012e-01 
          20    1.67010101996501e+00   1.06269628600354e+00   1.06269628600354e+00 
