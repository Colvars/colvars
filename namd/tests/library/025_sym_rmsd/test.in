colvarsTrajFrequency 1

colvar {
    name rmsd

    rmsd {
        atoms {
            centerReference yes
            rotateReference yes
            fittingGroup {
                atomNumbers 50 51 52 53 54 55 56 57 58
            }
            refPositionsFile ../Common/da.pdb
            atomNumbers 6 7 8 9
        }
        refPositionsFile ../Common/da_flipped.pdb
    }
}

colvar {
    name rmsd_sym

    rmsd {
        atoms {
            centerReference yes
            rotateReference yes
            fittingGroup {
                atomNumbers 50 51 52 53 54 55 56 57 58
            }
            refPositionsFile ../Common/da.pdb
            atomNumbers 6 7 8 9
        }
        refPositionsFile ../Common/da_flipped.pdb
        atomPermutation 6 8 9 7
        atomPermutation 6 9 7 8
    }
}

colvar {
    name rmsd_min

    customFunction min(min(r1, r2), r3)
    rmsd {
        name r1
        atoms {
            centerReference yes
            rotateReference yes
            fittingGroup {
                atomNumbers 50 51 52 53 54 55 56 57 58
            }
            refPositionsFile ../Common/da.pdb
            atomNumbers 6 7 8 9
        }
        refPositionsFile ../Common/da.pdb
    }
    rmsd {
        name r2
        atoms {
            centerReference yes
            rotateReference yes
            fittingGroup {
                atomNumbers 50 51 52 53 54 55 56 57 58
            }
            refPositionsFile ../Common/da.pdb
            atomNumbers 6 7 8 9
        }
        refPositionsFile ../Common/da_flipped.pdb
    }
    rmsd {
        name r3
        atoms {
            centerReference yes
            rotateReference yes
            fittingGroup {
                atomNumbers 50 51 52 53 54 55 56 57 58
            }
            refPositionsFile ../Common/da.pdb
            atomNumbers 6 7 8 9
        }
        refPositionsFile ../Common/da_flipped2.pdb
    }
}

harmonic {
  colvars rmsd_sym
  centers 0
  forceConstant 100
}
