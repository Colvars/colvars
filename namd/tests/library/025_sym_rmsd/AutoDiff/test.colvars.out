colvars: ----------------------------------------------------------------------
colvars: Please cite Fiorin et al, Mol Phys 2013:
colvars:  https://dx.doi.org/10.1080/00268976.2013.813594
colvars: in any publication based on this calculation.
colvars: SMP parallelism is enabled; if needed, use "smp off" to override this.
colvars: This version was built with the C++11 standard or higher.
colvars: ----------------------------------------------------------------------
colvars: Reading new configuration from file "test.in":
colvars: # units = "" [default]
colvars: # smp = on [default]
colvars: # colvarsTrajFrequency = 1
colvars: # colvarsRestartFrequency = 10 [default]
colvars: # scriptedColvarForces = off [default]
colvars: # scriptingAfterBiases = off [default]
colvars: ----------------------------------------------------------------------
colvars:   Initializing a new collective variable.
colvars:   # name = "rmsd"
colvars:   Initializing a new "rmsd" component.
colvars:     # name = "" [default]
colvars:     # componentCoeff = 1 [default]
colvars:     # componentExp = 1 [default]
colvars:     # period = 0 [default]
colvars:     # wrapAround = 0 [default]
colvars:     # forceNoPBC = off [default]
colvars:     # scalable = on [default]
colvars:       Initializing atom group "atoms".
colvars:       # name = "" [default]
colvars:       # centerReference = on
colvars:       # rotateReference = on
colvars:       # atomsOfGroup = "" [default]
colvars:       # indexGroup = "" [default]
colvars:       # psfSegID =  [default]
colvars:       # atomsFile = "" [default]
colvars:       # dummyAtom = ( 0 , 0 , 0 ) [default]
colvars:       # enableForces = on [default]
colvars:       Within atom group "atoms":
colvars:       Initializing atom group "fittingGroup".
colvars:       # name = "" [default]
colvars:       # centerReference = off [default]
colvars:       # rotateReference = off [default]
colvars:       # atomsOfGroup = "" [default]
colvars:       # indexGroup = "" [default]
colvars:       # psfSegID =  [default]
colvars:       # atomsFile = "" [default]
colvars:       # dummyAtom = ( 0 , 0 , 0 ) [default]
colvars:       # enableForces = on [default]
colvars:       # enableFitGradients = on [default]
colvars:       # printAtomIDs = off [default]
colvars:       Atom group "fittingGroup" defined with 9 atoms requested: total mass = 70.071, total charge = -0.09.
colvars:       # refPositions =  [default]
colvars:       # refPositionsFile = "../Common/da.pdb"
colvars:       # refPositionsCol = "" [default]
colvars:       Warning: atom group "atoms" will be aligned to a fixed orientation given by the reference positions provided.  If the internal structure of the group changes too much (i.e. its RMSD is comparable to its radius of gyration), the optimal rotation and its gradients may become discontinuous.  If that happens, use fittingGroup (or a different definition for it if already defined) to align the coordinates.
colvars:       # enableFitGradients = on [default]
colvars:       # printAtomIDs = off [default]
colvars:       Atom group "atoms" defined with 4 atoms requested: total mass = 15.035, total charge = 0.
colvars:     The option "fittingGroup" (alternative group for fitting) was enabled: Jacobian derivatives of the RMSD will not be calculated.
colvars:     # refPositions =  [default]
colvars:     # refPositionsFile = "../Common/da_flipped.pdb"
colvars:     # refPositionsCol = "" [default]
colvars:     WARNING: explicit fitting parameters were provided for atom group "atoms".
colvars:   All components initialized.
colvars:   # timeStepFactor = 1 [default]
colvars:   # width = 1 [default]
colvars:   # lowerBoundary = 0 [default]
colvars:   # upperBoundary = 0 [default]
colvars:   # hardLowerBoundary = on [default]
colvars:   # hardUpperBoundary = off [default]
colvars:   # expandBoundaries = off [default]
colvars:   # extendedLagrangian = off [default]
colvars:   # outputValue = on [default]
colvars:   # outputVelocity = off [default]
colvars:   # outputTotalForce = off [default]
colvars:   # outputAppliedForce = off [default]
colvars:   # subtractAppliedForce = off [default]
colvars:   # runAve = off [default]
colvars:   # corrFunc = off [default]
colvars: ----------------------------------------------------------------------
colvars:   Initializing a new collective variable.
colvars:   # name = "rmsd_sym"
colvars:   Initializing a new "rmsd" component.
colvars:     # name = "" [default]
colvars:     # componentCoeff = 1 [default]
colvars:     # componentExp = 1 [default]
colvars:     # period = 0 [default]
colvars:     # wrapAround = 0 [default]
colvars:     # forceNoPBC = off [default]
colvars:     # scalable = on [default]
colvars:       Initializing atom group "atoms".
colvars:       # name = "" [default]
colvars:       # centerReference = on
colvars:       # rotateReference = on
colvars:       # atomsOfGroup = "" [default]
colvars:       # indexGroup = "" [default]
colvars:       # psfSegID =  [default]
colvars:       # atomsFile = "" [default]
colvars:       # dummyAtom = ( 0 , 0 , 0 ) [default]
colvars:       # enableForces = on [default]
colvars:       Within atom group "atoms":
colvars:       Initializing atom group "fittingGroup".
colvars:       # name = "" [default]
colvars:       # centerReference = off [default]
colvars:       # rotateReference = off [default]
colvars:       # atomsOfGroup = "" [default]
colvars:       # indexGroup = "" [default]
colvars:       # psfSegID =  [default]
colvars:       # atomsFile = "" [default]
colvars:       # dummyAtom = ( 0 , 0 , 0 ) [default]
colvars:       # enableForces = on [default]
colvars:       # enableFitGradients = on [default]
colvars:       # printAtomIDs = off [default]
colvars:       Atom group "fittingGroup" defined with 9 atoms requested: total mass = 70.071, total charge = -0.09.
colvars:       # refPositions =  [default]
colvars:       # refPositionsFile = "../Common/da.pdb"
colvars:       # refPositionsCol = "" [default]
colvars:       Warning: atom group "atoms" will be aligned to a fixed orientation given by the reference positions provided.  If the internal structure of the group changes too much (i.e. its RMSD is comparable to its radius of gyration), the optimal rotation and its gradients may become discontinuous.  If that happens, use fittingGroup (or a different definition for it if already defined) to align the coordinates.
colvars:       # enableFitGradients = on [default]
colvars:       # printAtomIDs = off [default]
colvars:       Atom group "atoms" defined with 4 atoms requested: total mass = 15.035, total charge = 0.
colvars:     The option "fittingGroup" (alternative group for fitting) was enabled: Jacobian derivatives of the RMSD will not be calculated.
colvars:     # refPositions =  [default]
colvars:     # refPositionsFile = "../Common/da_flipped.pdb"
colvars:     # refPositionsCol = "" [default]
colvars:     WARNING: explicit fitting parameters were provided for atom group "atoms".
colvars:     atomPermutation = { 0, 2, 3, 1 }
colvars:     atomPermutation = { 0, 3, 1, 2 }
colvars:   All components initialized.
colvars:   # timeStepFactor = 1 [default]
colvars:   # width = 1 [default]
colvars:   # lowerBoundary = 0 [default]
colvars:   # upperBoundary = 0 [default]
colvars:   # hardLowerBoundary = on [default]
colvars:   # hardUpperBoundary = off [default]
colvars:   # expandBoundaries = off [default]
colvars:   # extendedLagrangian = off [default]
colvars:   # outputValue = on [default]
colvars:   # outputVelocity = off [default]
colvars:   # outputTotalForce = off [default]
colvars:   # outputAppliedForce = off [default]
colvars:   # subtractAppliedForce = off [default]
colvars:   # runAve = off [default]
colvars:   # corrFunc = off [default]
colvars: ----------------------------------------------------------------------
colvars:   Initializing a new collective variable.
colvars:   # name = "rmsd_min"
colvars:   Initializing a new "rmsd" component.
colvars:     # name = "r1"
colvars:     # componentCoeff = 1 [default]
colvars:     # componentExp = 1 [default]
colvars:     # period = 0 [default]
colvars:     # wrapAround = 0 [default]
colvars:     # forceNoPBC = off [default]
colvars:     # scalable = on [default]
colvars:       Initializing atom group "atoms".
colvars:       # name = "" [default]
colvars:       # centerReference = on
colvars:       # rotateReference = on
colvars:       # atomsOfGroup = "" [default]
colvars:       # indexGroup = "" [default]
colvars:       # psfSegID =  [default]
colvars:       # atomsFile = "" [default]
colvars:       # dummyAtom = ( 0 , 0 , 0 ) [default]
colvars:       # enableForces = on [default]
colvars:       Within atom group "atoms":
colvars:       Initializing atom group "fittingGroup".
colvars:       # name = "" [default]
colvars:       # centerReference = off [default]
colvars:       # rotateReference = off [default]
colvars:       # atomsOfGroup = "" [default]
colvars:       # indexGroup = "" [default]
colvars:       # psfSegID =  [default]
colvars:       # atomsFile = "" [default]
colvars:       # dummyAtom = ( 0 , 0 , 0 ) [default]
colvars:       # enableForces = on [default]
colvars:       # enableFitGradients = on [default]
colvars:       # printAtomIDs = off [default]
colvars:       Atom group "fittingGroup" defined with 9 atoms requested: total mass = 70.071, total charge = -0.09.
colvars:       # refPositions =  [default]
colvars:       # refPositionsFile = "../Common/da.pdb"
colvars:       # refPositionsCol = "" [default]
colvars:       Warning: atom group "atoms" will be aligned to a fixed orientation given by the reference positions provided.  If the internal structure of the group changes too much (i.e. its RMSD is comparable to its radius of gyration), the optimal rotation and its gradients may become discontinuous.  If that happens, use fittingGroup (or a different definition for it if already defined) to align the coordinates.
colvars:       # enableFitGradients = on [default]
colvars:       # printAtomIDs = off [default]
colvars:       Atom group "atoms" defined with 4 atoms requested: total mass = 15.035, total charge = 0.
colvars:     The option "fittingGroup" (alternative group for fitting) was enabled: Jacobian derivatives of the RMSD will not be calculated.
colvars:     # refPositions =  [default]
colvars:     # refPositionsFile = "../Common/da.pdb"
colvars:     # refPositionsCol = "" [default]
colvars:     WARNING: explicit fitting parameters were provided for atom group "atoms".
colvars:   Initializing a new "rmsd" component.
colvars:     # name = "r2"
colvars:     # componentCoeff = 1 [default]
colvars:     # componentExp = 1 [default]
colvars:     # period = 0 [default]
colvars:     # wrapAround = 0 [default]
colvars:     # forceNoPBC = off [default]
colvars:     # scalable = on [default]
colvars:       Initializing atom group "atoms".
colvars:       # name = "" [default]
colvars:       # centerReference = on
colvars:       # rotateReference = on
colvars:       # atomsOfGroup = "" [default]
colvars:       # indexGroup = "" [default]
colvars:       # psfSegID =  [default]
colvars:       # atomsFile = "" [default]
colvars:       # dummyAtom = ( 0 , 0 , 0 ) [default]
colvars:       # enableForces = on [default]
colvars:       Within atom group "atoms":
colvars:       Initializing atom group "fittingGroup".
colvars:       # name = "" [default]
colvars:       # centerReference = off [default]
colvars:       # rotateReference = off [default]
colvars:       # atomsOfGroup = "" [default]
colvars:       # indexGroup = "" [default]
colvars:       # psfSegID =  [default]
colvars:       # atomsFile = "" [default]
colvars:       # dummyAtom = ( 0 , 0 , 0 ) [default]
colvars:       # enableForces = on [default]
colvars:       # enableFitGradients = on [default]
colvars:       # printAtomIDs = off [default]
colvars:       Atom group "fittingGroup" defined with 9 atoms requested: total mass = 70.071, total charge = -0.09.
colvars:       # refPositions =  [default]
colvars:       # refPositionsFile = "../Common/da.pdb"
colvars:       # refPositionsCol = "" [default]
colvars:       Warning: atom group "atoms" will be aligned to a fixed orientation given by the reference positions provided.  If the internal structure of the group changes too much (i.e. its RMSD is comparable to its radius of gyration), the optimal rotation and its gradients may become discontinuous.  If that happens, use fittingGroup (or a different definition for it if already defined) to align the coordinates.
colvars:       # enableFitGradients = on [default]
colvars:       # printAtomIDs = off [default]
colvars:       Atom group "atoms" defined with 4 atoms requested: total mass = 15.035, total charge = 0.
colvars:     The option "fittingGroup" (alternative group for fitting) was enabled: Jacobian derivatives of the RMSD will not be calculated.
colvars:     # refPositions =  [default]
colvars:     # refPositionsFile = "../Common/da_flipped.pdb"
colvars:     # refPositionsCol = "" [default]
colvars:     WARNING: explicit fitting parameters were provided for atom group "atoms".
colvars:   Initializing a new "rmsd" component.
colvars:     # name = "r3"
colvars:     # componentCoeff = 1 [default]
colvars:     # componentExp = 1 [default]
colvars:     # period = 0 [default]
colvars:     # wrapAround = 0 [default]
colvars:     # forceNoPBC = off [default]
colvars:     # scalable = on [default]
colvars:       Initializing atom group "atoms".
colvars:       # name = "" [default]
colvars:       # centerReference = on
colvars:       # rotateReference = on
colvars:       # atomsOfGroup = "" [default]
colvars:       # indexGroup = "" [default]
colvars:       # psfSegID =  [default]
colvars:       # atomsFile = "" [default]
colvars:       # dummyAtom = ( 0 , 0 , 0 ) [default]
colvars:       # enableForces = on [default]
colvars:       Within atom group "atoms":
colvars:       Initializing atom group "fittingGroup".
colvars:       # name = "" [default]
colvars:       # centerReference = off [default]
colvars:       # rotateReference = off [default]
colvars:       # atomsOfGroup = "" [default]
colvars:       # indexGroup = "" [default]
colvars:       # psfSegID =  [default]
colvars:       # atomsFile = "" [default]
colvars:       # dummyAtom = ( 0 , 0 , 0 ) [default]
colvars:       # enableForces = on [default]
colvars:       # enableFitGradients = on [default]
colvars:       # printAtomIDs = off [default]
colvars:       Atom group "fittingGroup" defined with 9 atoms requested: total mass = 70.071, total charge = -0.09.
colvars:       # refPositions =  [default]
colvars:       # refPositionsFile = "../Common/da.pdb"
colvars:       # refPositionsCol = "" [default]
colvars:       Warning: atom group "atoms" will be aligned to a fixed orientation given by the reference positions provided.  If the internal structure of the group changes too much (i.e. its RMSD is comparable to its radius of gyration), the optimal rotation and its gradients may become discontinuous.  If that happens, use fittingGroup (or a different definition for it if already defined) to align the coordinates.
colvars:       # enableFitGradients = on [default]
colvars:       # printAtomIDs = off [default]
colvars:       Atom group "atoms" defined with 4 atoms requested: total mass = 15.035, total charge = 0.
colvars:     The option "fittingGroup" (alternative group for fitting) was enabled: Jacobian derivatives of the RMSD will not be calculated.
colvars:     # refPositions =  [default]
colvars:     # refPositionsFile = "../Common/da_flipped2.pdb"
colvars:     # refPositionsCol = "" [default]
colvars:     WARNING: explicit fitting parameters were provided for atom group "atoms".
colvars:   All components initialized.
colvars:   This colvar uses a custom function.
colvars:   Expecting colvar value of type scalar number.
colvars:   # period = 0 [default]
colvars:   # timeStepFactor = 1 [default]
colvars:   # width = 1 [default]
colvars:   # lowerBoundary = 0 [default]
colvars:   # upperBoundary = 0 [default]
colvars:   # hardLowerBoundary = off [default]
colvars:   # hardUpperBoundary = off [default]
colvars:   # expandBoundaries = off [default]
colvars:   # extendedLagrangian = off [default]
colvars:   # outputValue = on [default]
colvars:   # outputVelocity = off [default]
colvars:   # outputTotalForce = off [default]
colvars:   # outputAppliedForce = off [default]
colvars:   # subtractAppliedForce = off [default]
colvars:   # runAve = off [default]
colvars:   # corrFunc = off [default]
colvars: ----------------------------------------------------------------------
colvars: Collective variables initialized, 3 in total.
colvars: ----------------------------------------------------------------------
colvars:   Initializing a new "harmonic" instance.
colvars:   # name = "harmonic1" [default]
colvars:   # colvars = { rmsd_sym }
colvars:   # outputEnergy = off [default]
colvars:   # outputFreq = 10 [default]
colvars:   # timeStepFactor = 1 [default]
colvars:   # writeTISamples = off [default]
colvars:   # writeTIPMF = off [default]
colvars:   # centers = { 0 }
colvars:   # targetCenters = { 0 } [default]
colvars:   # outputCenters = off [default]
colvars:   # forceConstant = 100
colvars:   # targetForceConstant = -1 [default]
colvars:   The force constant for colvar "rmsd_sym" will be rescaled to 100 according to the specified width (1).
colvars: ----------------------------------------------------------------------
colvars: Collective variables biases initialized, 1 in total.
colvars: ----------------------------------------------------------------------
colvars: Collective variables module (re)initialized.
colvars: ----------------------------------------------------------------------
colvars: Updating NAMD interface:
colvars: updating atomic data (13 atoms).
colvars: updating group data (0 scalable groups, 0 atoms in total).
colvars: Re-initialized atom group for variable "rmsd":0/0. 4 atoms: total mass = 15.035, total charge = 0.
colvars: Re-initialized atom group for variable "rmsd_sym":0/0. 4 atoms: total mass = 15.035, total charge = 0.
colvars: Re-initialized atom group for variable "rmsd_min":0/0. 4 atoms: total mass = 15.035, total charge = 0.
colvars: Re-initialized atom group for variable "rmsd_min":1/0. 4 atoms: total mass = 15.035, total charge = 0.
colvars: Re-initialized atom group for variable "rmsd_min":2/0. 4 atoms: total mass = 15.035, total charge = 0.
colvars: The restart output state file will be "test.tmp.colvars.state".
colvars: The final output state file will be "test.colvars.state".
colvars: Opening trajectory file "test.colvars.traj".
colvars: Redefining the Tcl "cv" command to the new script interface.
colvars: Updating NAMD interface:
colvars: updating atomic data (13 atoms).
colvars: updating group data (0 scalable groups, 0 atoms in total).
colvars: Re-initialized atom group for variable "rmsd":0/0. 4 atoms: total mass = 15.035, total charge = 0.
colvars: Re-initialized atom group for variable "rmsd_sym":0/0. 4 atoms: total mass = 15.035, total charge = 0.
colvars: Re-initialized atom group for variable "rmsd_min":0/0. 4 atoms: total mass = 15.035, total charge = 0.
colvars: Re-initialized atom group for variable "rmsd_min":1/0. 4 atoms: total mass = 15.035, total charge = 0.
colvars: Re-initialized atom group for variable "rmsd_min":2/0. 4 atoms: total mass = 15.035, total charge = 0.
colvars: The restart output state file will be "test.tmp.colvars.state".
colvars: The final output state file will be "test.colvars.state".
colvars: Synchronizing (emptying the buffer of) trajectory file "test.colvars.traj".
colvars: Synchronizing (emptying the buffer of) trajectory file "test.colvars.traj".
colvars: Saving collective variables state to "test.tmp.colvars.state".
colvars: Synchronizing (emptying the buffer of) trajectory file "test.colvars.traj".
colvars: Saving collective variables state to "test.tmp.colvars.state".
colvars: Saving collective variables state to "test.colvars.state".
